save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationToolsMenu.kt

```diff
@@ -68,11 +68,13 @@ private fun organizationToolsMenu() = FC<OrganizationToolsMenuProps> { props ->
                 .thenBy { it.name }
 
     /**
+     * This function delete [project] by [projects], add [project] with [newStatus] in [projects] and sorted its by comparator
```

Comment:
It seems to be too complicated still, maybe you are doing something wrong as far as if it is hard to tell what does the function do, then something is wrong with it

And please do not make it be inside another function. Either make it lambda or move it away

Reply:
agree, maybe change signature to
```kotlin
fun updateProjects(oldProject: ProjectDto, newProject: ProjectDto) {
    projects
        .minus(oldProject).plus(newProject)
        .sortedWith(comparator)
        .also {
            setProjects(it)
        }
        .also {
            props.updateProjects(it)
        }
}
```

And then it can be called like:
```kotlin
updateProjects(
    oldProject = project,
    newProject = project.copy(
        status = if (isBanMode) ProjectStatus.BANNED else ProjectStatus.DELETED
    )
)
```

Reply:
Done